### ISSUE TITLE
possible problem with wicket 7.2

### ISSUE THREAD

**Issue by nmwael** [l0rdn1kk0n/wicket-bootstrap#590]
Using bootstrap modal with a form within gives error messages about missing fields.

**Comment by nmwael** [comment]
Unexpected RuntimeException
Last cause: Unable to find component with id 'accountNumber' in [Form [Component id = addContactForm]]
	Expected: 'campaignDetailFragmentContainer:campaignDetailForm:addContactPanel:addContactForm:accountNumber'.
	Found with similar names: ''

Markup
The problem is in "file:/C:/nino/develop/workspace/dialer/webdialer/target/classes/dk/netdesign/ekspresbank/panel/AddContactPanel.html":

<form class="form-horizontal" role="form" wicket:id="addContactForm">
                        <div class="form-group" > 
                            <label class="col-sm-6">accountNumber</label>
                            <div class="col-sm-6">
                                <input class="form-control" data-toggle="tooltip" title="Contact account number" type="text" wicket:id="accountNumber"/>
                            </div>
                        </div>
                        <div class="form-group"> 
                            <label class="col-sm-6">queueNumber</label>
                            <div class="col-sm-6">
                                <input class="form-control" data-toggle="tooltip" title="Contact queueNumber" type="text" wicket:id="queueNumber"/>
                            </div>
                        </div>
                        <div class="form-group"> 
                            <label class="col-sm-6">balance</label>
                            <div class="col-sm-6">
                                <input class="form-control" data-toggle="tooltip" title="Contact balance" type="text" wicket:id="balance"/>
                            </div>
                        </div>
                        <div class="form-group"> 
                            <label class="col-sm-6">personalIdentificationNumber</label>
                            <div class="col-sm-6">
                                <input class="form-control" data-toggle="tooltip" title="Contact balance" type="text" wicket:id="personalIdentificationNumber"/>
                            </div>
                        </div>
                        <div class="form-group"> 
                            <label class="col-sm-6">firstName</label>
                            <div class="col-sm-6">
                                <input class="form-control" data-toggle="tooltip" title="Contact first name" type="text" wicket:id="firstName"/>
                            </div>
                        </div>
                        <div class="form-group"> 
                            <label class="col-sm-6">lastName</label>
                            <div class="col-sm-6">
                                <input class="form-control" data-toggle="tooltip" title="Contact last name" type="text" wicket:id="lastName"/>
                            </div>
                        </div>
                        <div class="form-group"> 
                            <label class="col-sm-6">address</label>
                            <div class="col-sm-6">
                                <input class="form-control" data-toggle="tooltip" title="Contact address" type="text" wicket:id="address"/>
                            </div>
                        </div>
                        <div class="form-group"> 
                            <label class="col-sm-6">address</label>
                            <div class="col-sm-6">
                                <input class="form-control" data-toggle="tooltip" title="Contact zipCode" type="text" wicket:id="zipCode"/>
                            </div>
                        </div>
                        <div class="form-group"> 
                            <label class="col-sm-6">city</label>
                            <div class="col-sm-6">
                                <input class="form-control" data-toggle="tooltip" title="Contact city" type="text" wicket:id="city"/>
                            </div>
                        </div>
                        <div class="form-group"> 
                            <label class="col-sm-6">homePhone</label>
                            <div class="col-sm-6">
                                <input class="form-control" data-toggle="tooltip" title="Contact homePhone" type="text" wicket:id="homePhone"/>
                            </div>
                        </div>
                        <div class="form-group"> 
                            <label class="col-sm-6">mobilePhone</label>
                            <div class="col-sm-6">
                                <input class="form-control" data-toggle="tooltip" title="Contact mobilePhone" type="text" wicket:id="mobilePhone"/>
                            </div>
                        </div>
                    </form>

Stacktrace
Root cause:

Unable to find component with id 'accountNumber' in [Form [Component id = addContactForm]]
	Expected: 'campaignDetailFragmentContainer:campaignDetailForm:addContactPanel:addContactForm:accountNumber'.
	Found with similar names: ''
 MarkupStream: [markup = file:/C:/nino/develop/workspace/dialer/webdialer/target/classes/dk/netdesign/ekspresbank/panel/AddContactPanel.html
<form class="form-horizontal" role="form" wicket:id="addContactForm">
                        <div class="form-group" > 
                            <label class="col-sm-6">accountNumber</label>
                            <div class="col-sm-6">
                                <input class="form-control" data-toggle="tooltip" title="Contact account number" type="text" wicket:id="accountNumber"/>
                            </div>
                        </div>
                        <div class="form-group"> 
                            <label class="col-sm-6">queueNumber</label>
                            <div class="col-sm-6">
                                <input class="form-control" data-toggle="tooltip" title="Contact queueNumber" type="text" wicket:id="queueNumber"/>
                            </div>
                        </div>
                        <div class="form-group"> 
                            <label class="col-sm-6">balance</label>
                            <div class="col-sm-6">
                                <input class="form-control" data-toggle="tooltip" title="Contact balance" type="text" wicket:id="balance"/>
                            </div>
                        </div>
                        <div class="form-group"> 
                            <label class="col-sm-6">personalIdentificationNumber</label>
                            <div class="col-sm-6">
                                <input class="form-control" data-toggle="tooltip" title="Contact balance" type="text" wicket:id="personalIdentificationNumber"/>
                            </div>
                        </div>
                        <div class="form-group"> 
                            <label class="col-sm-6">firstName</label>
                            <div class="col-sm-6">
                                <input class="form-control" data-toggle="tooltip" title="Contact first name" type="text" wicket:id="firstName"/>
                            </div>
                        </div>
                        <div class="form-group"> 
                            <label class="col-sm-6">lastName</label>
                            <div class="col-sm-6">
                                <input class="form-control" data-toggle="tooltip" title="Contact last name" type="text" wicket:id="lastName"/>
                            </div>
                        </div>
                        <div class="form-group"> 
                            <label class="col-sm-6">address</label>
                            <div class="col-sm-6">
                                <input class="form-control" data-toggle="tooltip" title="Contact address" type="text" wicket:id="address"/>
                            </div>
                        </div>
                        <div class="form-group"> 
                            <label class="col-sm-6">address</label>
                            <div class="col-sm-6">
                                <input class="form-control" data-toggle="tooltip" title="Contact zipCode" type="text" wicket:id="zipCode"/>
                            </div>
                        </div>
                        <div class="form-group"> 
                            <label class="col-sm-6">city</label>
                            <div class="col-sm-6">
                                <input class="form-control" data-toggle="tooltip" title="Contact city" type="text" wicket:id="city"/>
                            </div>
                        </div>
                        <div class="form-group"> 
                            <label class="col-sm-6">homePhone</label>
                            <div class="col-sm-6">
                                <input class="form-control" data-toggle="tooltip" title="Contact homePhone" type="text" wicket:id="homePhone"/>
                            </div>
                        </div>
                        <div class="form-group"> 
                            <label class="col-sm-6">mobilePhone</label>
                            <div class="col-sm-6">
                                <input class="form-control" data-toggle="tooltip" title="Contact mobilePhone" type="text" wicket:id="mobilePhone"/>
                            </div>
                        </div>
                    </form>, index = 2, current =  '<input class="form-control" data-toggle="tooltip" title="Contact account number" type="text" wicket:id="accountNumber"/>' (line 0, column 0)]
     at org.apache.wicket.markup.MarkupStream.throwMarkupException(MarkupStream.java:526)
     at org.apache.wicket.MarkupContainer.throwException(MarkupContainer.java:1618)
     at org.apache.wicket.MarkupContainer.renderNext(MarkupContainer.java:1537)
     at org.apache.wicket.MarkupContainer.renderAll(MarkupContainer.java:1759)
     at org.apache.wicket.MarkupContainer.renderComponentTagBody(MarkupContainer.java:1734)
     at org.apache.wicket.MarkupContainer.onComponentTagBody(MarkupContainer.java:1657)
     at org.apache.wicket.markup.html.form.Form.onComponentTagBody(Form.java:1705)
     at org.apache.wicket.markup.html.panel.DefaultMarkupSourcingStrategy.onComponentTagBody(DefaultMarkupSourcingStrategy.java:70)
     at org.apache.wicket.Component.internalRenderComponent(Component.java:2565)
     at org.apache.wicket.MarkupContainer.onRender(MarkupContainer.java:1698)
     at org.apache.wicket.Component.internalRender(Component.java:2380)
     at org.apache.wicket.Component.render(Component.java:2308)
     at org.apache.wicket.MarkupContainer.renderNext(MarkupContainer.java:1524)
     at org.apache.wicket.MarkupContainer.renderAll(MarkupContainer.java:1759)
     at org.apache.wicket.MarkupContainer.renderComponentTagBody(MarkupContainer.java:1734)
     at org.apache.wicket.MarkupContainer.onComponentTagBody(MarkupContainer.java:1657)
     at org.apache.wicket.markup.html.panel.DefaultMarkupSourcingStrategy.onComponentTagBody(DefaultMarkupSourcingStrategy.java:70)
     at org.apache.wicket.Component.internalRenderComponent(Component.java:2565)
     at org.apache.wicket.MarkupContainer.onRender(MarkupContainer.java:1698)
     at org.apache.wicket.Component.internalRender(Component.java:2380)
     at org.apache.wicket.Component.render(Component.java:2308)
     at org.apache.wicket.MarkupContainer.renderNext(MarkupContainer.java:1524)
     at org.apache.wicket.MarkupContainer.renderAll(MarkupContainer.java:1759)
     at org.apache.wicket.MarkupContainer.renderComponentTagBody(MarkupContainer.java:1734)
     at org.apache.wicket.MarkupContainer.renderAssociatedMarkup(MarkupContainer.java:797)
     at org.apache.wicket.markup.html.panel.AssociatedMarkupSourcingStrategy.renderAssociatedMarkup(AssociatedMarkupSourcingStrategy.java:77)
     at org.apache.wicket.markup.html.panel.PanelMarkupSourcingStrategy.onComponentTagBody(PanelMarkupSourcingStrategy.java:112)
     at org.apache.wicket.Component.internalRenderComponent(Component.java:2565)
     at org.apache.wicket.MarkupContainer.onRender(MarkupContainer.java:1698)
     at org.apache.wicket.Component.internalRender(Component.java:2380)
     at org.apache.wicket.Component.render(Component.java:2308)
     at org.apache.wicket.page.XmlPartialPageUpdate.writeComponent(XmlPartialPageUpdate.java:119)
     at org.apache.wicket.page.PartialPageUpdate.writeComponents(PartialPageUpdate.java:245)
     at org.apache.wicket.page.PartialPageUpdate.writeTo(PartialPageUpdate.java:158)
     at org.apache.wicket.ajax.AjaxRequestHandler.respond(AjaxRequestHandler.java:358)
     at org.apache.wicket.request.cycle.RequestCycle$HandlerExecutor.respond(RequestCycle.java:895)
     at org.apache.wicket.request.RequestHandlerStack.execute(RequestHandlerStack.java:64)
     at org.apache.wicket.request.RequestHandlerStack.execute(RequestHandlerStack.java:97)
     at org.apache.wicket.request.cycle.RequestCycle.execute(RequestCycle.java:265)
     at org.apache.wicket.request.cycle.RequestCycle.processRequest(RequestCycle.java:222)
     at org.apache.wicket.request.cycle.RequestCycle.processRequestAndDetach(RequestCycle.java:293)
     at org.apache.wicket.protocol.http.WicketFilter.processRequestCycle(WicketFilter.java:261)
     at org.apache.wicket.protocol.http.WicketFilter.processRequest(WicketFilter.java:203)
     at org.apache.wicket.protocol.http.WicketFilter.doFilter(WicketFilter.java:284)
     at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
     at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:585)
     at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
     at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:577)
     at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:223)
     at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1127)
     at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515)
     at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
     at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1061)
     at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
     at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
     at org.eclipse.jetty.server.Server.handle(Server.java:499)
     at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:310)
     at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:257)
     at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:540)
     at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635)
     at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555)
     at java.lang.Thread.run(Thread.java:745)

display page view

**Comment by martin-g** [comment]
I hope you realize that you don't provide any meaningful description of the
problem.

**Comment by nmwael** [comment]
If have a class that extends de.agilecoders.wicket.core.markup.html.bootstrap.dialog.Modal
when calling 
```
                exclusionPanel.setVisible(true);
                target.add(exclusionPanel);
                exclusionPanel.show(target);

```

I suddently get the error mentioned above, it was working with wicket 7.1

**Comment by tahitiangabriel** [comment]
Which version of wicket-bootstrap are you using? The latest?

**Comment by nmwael** [comment]
0.10.6

**Comment by martin-g** [comment]
Do you use enclosures?
They are a bit unstable due to the component queueing.
Please create a quickstart and attach it in Wicket JIRA.

**Comment by nmwael** [comment]
no enclosures... And will look into creating a quickstart.. But we are at final stage going into production monday.. 

SO i'll take the quick way out and do wicket modal for now.
